src/store/modules/product/actions.ts

```diff
@@ -9,43 +9,75 @@ import emitter from '@/event-bus'
 
 
 const actions: ActionTree<ProductState, RootState> = {
-
-  // Find Product
   async findProduct ({ commit, state }, payload) {
-
-    // Show loader only when new query and not the infinite scroll
-    if (payload.viewIndex === 0) emitter.emit("presentLoader");
-
     let resp;
-
     try {
       resp = await ProductService.fetchProducts({
-        // used sku as we are currently only using sku to search for the product
-        "filters": ['sku: ' + payload.queryString],
         "viewSize": payload.viewSize,
-        "viewIndex": payload.viewIndex
+        "viewIndex": payload.viewIndex,
+        "facilityId": payload.facilityId,
+        "statusId": "PURCH_SHIP_SHIPPED"
       })
-
-      // resp.data.response.numFound tells the number of items in the response
-      if (resp.status === 200 && resp.data.response.numFound > 0 && !hasError(resp)) {
-        let products = resp.data.response.docs;
-        const totalProductsCount = resp.data.response.numFound;
+      if (resp.status === 200 && resp.data.count> 0 && !hasError(resp)) {
+        let products = resp.data.docs;
 
         if (payload.viewIndex && payload.viewIndex > 0) products = state.products.list.concat(products)
-        commit(types.PRODUCT_SEARCH_UPDATED, { products: products, totalProductsCount: totalProductsCount })
+        commit(types.PRODUCT_SEARCH_UPDATED, { products: products })
       } else {
-        //showing error whenever getting no products in the response or having any other error
         showToast(translate("Product not found"));
       }
-      // Remove added loader only when new query and not the infinite scroll
       if (payload.viewIndex === 0) emitter.emit("dismissLoader");
     } catch(error){
       console.log(error)
       showToast(translate("Something went wrong"));
     }
-    // TODO Handle specific error
     return resp;
   },
+  async setCurrentProduct ({ commit }, payload) {
+    let resp;
+
+    try {
+      resp = await ProductService.getShipmentProduct({
+        "shipmentId": payload.shipmentId,
+      });
+      if (resp.status === 200 && resp.data.items&& !hasError(resp)) {
+        commit(types.PRODUCT_CURRENT, { current: resp.data })
+        return resp.data;
+      } else {
+        showToast(translate('Something went wrong'));
+        console.error("error", resp.data._ERROR_MESSAGE_);
+        return Promise.reject(new Error(resp.data._ERROR_MESSAGE_));
+      }
+
+    } catch (err) {
+      showToast(translate('Something went wrong'));
+      console.error("error", err);
+      return Promise.reject(new Error(err))
+    }
+  },
+  async updateShipmentProducts({ commit }, payload) {
+    emitter.emit("presentLoader");
+    let resp;
+    try {
+      resp = await ProductService.receiveShipmentItems(payload);
```

Comment:
I think we should change the name to ShipmentService instead of ProductService, WDYT?

Reply:
Done